src/mbgl/map/map.cpp

```diff
@@ -997,6 +1004,22 @@ void Map::Impl::onUpdate(Update flags) {
     
 void Map::Impl::onStyleLoaded() {
     view.notifyMapChange(MapChangeDidFinishLoadingStyle);
+    
+    if (initialCameraOptions == nullptr) {
+        // user manipulated camera before style was loaded
+        return;
+    }
+    
+    // move camera to default camera of style
+    initialCameraOptions->center = style->getDefaultLatLng();
+    initialCameraOptions->zoom = style->getDefaultZoom();
+    initialCameraOptions->angle = style->getDefaultBearing();
+    initialCameraOptions->pitch = style->getDefaultPitch();
+    transform.jumpTo(*initialCameraOptions);
+}
+
+void Map::Impl::setIgnoreInitialCameraOptions() {
```

Comment:
Inline this method. Seeing `initialCameraOptions` get cleared out is pretty self-explanatory; no need for a separate method.